aiohue/v2/controllers/devices.py

```diff
@@ -19,17 +19,11 @@ class DevicesController(BaseResourcesController[Type[Device]]):
 
     def get_lights(self, id: str) -> List[Light]:
         """Return all lights belonging to given device."""
-        return [
-            self._bridge.lights[x] for x in self[id].lights if x in self._bridge.lights
-        ]
+        return [x for x in self._bridge.lights if x.id in self[id].lights]
```

Comment:
you're only returning `x` here, which I think is the key. However it's typed as returning the light object. Is this a breaking change?

Reply:
Good catch, that is actually a typo. I'll fix it

Reply:
Actually the code is fine and the actual light object is returned here so all good...

<img width="673" alt="Schermafbeelding 2022-02-24 om 20 41 15" src="https://user-images.githubusercontent.com/6389780/155595472-2b313e98-3fb1-4d78-b34d-e9fa78387aa0.png">

Reply:
tested the code in HA too to see if it fixes the reported issue when adding a new light and it's all fine now